extensions/pkg/webhook/utils_test.go

```diff
@@ -0,0 +1,682 @@
+					Mode:          &modeAuto,
+				}
+				result := webhook.EnsureVPAContainerResourcePolicyWithName(policies, newPolicy)
+				Expect(result).To(Equal(append(policies, newPolicy)))
+			})
+
+			It("should replace the existing ContainerResourcePolicy if it's not identical", func() {
+				existingPolicy := vpaautoscalingv1.ContainerResourcePolicy{
+					ContainerName: "container1",
+					Mode:          &modeOff,
+				}
+				result := webhook.EnsureVPAContainerResourcePolicyWithName(policies, existingPolicy)
+				Expect(result).To(Equal([]vpaautoscalingv1.ContainerResourcePolicy{
+					{
+						ContainerName: "container1",
+						Mode:          &modeOff,
+					},
+					{
+						ContainerName: "container2",
+						Mode:          &modeOff,
+					},
+				}))
+			})
+
+			It("should do nothing to the existing ContainerResourcePolicy if it's identical", func() {
+				identicalPolicy := policies[0]
+				result := webhook.EnsureVPAContainerResourcePolicyWithName(policies, identicalPolicy)
+				Expect(result).To(Equal(policies))
+			})
+		})
+
+		Describe("EnsurePVCWithName", func() {
+			var pvcs []corev1.PersistentVolumeClaim
+
+			BeforeEach(func() {
+				pvcs = []corev1.PersistentVolumeClaim{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "pvc1",
+							Namespace: "namespace1",
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "pvc2",
+							Namespace: "namespace2",
+						},
+					},
+				}
+			})
+
+			It("should add a new PersistentVolumeClaim if not present", func() {
+				newPVC := corev1.PersistentVolumeClaim{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "pvc3",
+						Namespace: "namespace3",
+					},
+				}
+				result := webhook.EnsurePVCWithName(pvcs, newPVC)
+				Expect(result).To(Equal(append(pvcs, newPVC)))
+			})
+
+			It("should replace the existing PersistentVolumeClaim if it's not identical", func() {
+				existingPVC := corev1.PersistentVolumeClaim{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "pvc1",
+						Namespace: "namespace3",
+					},
+				}
+				result := webhook.EnsurePVCWithName(pvcs, existingPVC)
+				Expect(result).To(Equal([]corev1.PersistentVolumeClaim{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "pvc1",
+							Namespace: "namespace3",
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "pvc2",
+							Namespace: "namespace2",
+						},
+					},
+				}))
+			})
+
+			It("should do nothing to the existing PersistentVolumeClaim if it's identical", func() {
+				identicalPVC := pvcs[0]
+				result := webhook.EnsurePVCWithName(pvcs, identicalPVC)
+				Expect(result).To(Equal(pvcs))
+			})
+		})
+
+		Describe("EnsureFileWithPath", func() {
+			var files []extensionsv1alpha1.File
+			var permissions0644 int32 = 0644
+			var permissions0400 int32 = 0400
+
+			BeforeEach(func() {
+				files = []extensionsv1alpha1.File{
+					{
+						Path:        "path1",
+						Permissions: &permissions0644,
```

Comment:
I rather see that other unit tests rather use pointer.Int32() for setting the permissions.

```suggestion
						Permissions: pointer.Int32(0644),
```